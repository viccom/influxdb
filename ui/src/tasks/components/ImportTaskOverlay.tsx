import React, {PureComponent} from 'react'
import _ from 'lodash'

import Container from 'src/clockface/components/overlays/OverlayContainer'
import Heading from 'src/clockface/components/overlays/OverlayHeading'
import Body from 'src/clockface/components/overlays/OverlayBody'
import DragAndDrop from 'src/shared/components/DragAndDrop'

interface Props {
  onDismissOverlay: () => void
  onSave: (script: string, fileName: string) => void
}

class ImportTaskOverlay extends PureComponent<Props> {
  constructor(props: Props) {
    super(props)
  }

  public render() {
    const {onDismissOverlay} = this.props

    return (
      <Container maxWidth={800}>
        <Heading title="Import Task" onDismiss={onDismissOverlay} />
        <Body>
          <DragAndDrop
            submitText="Upload Task"
            fileTypesToAccept={this.validFileExtension}
            handleSubmit={this.handleUploadTask}
          />
        </Body>
      </Container>
    )
  }

  private get validFileExtension(): string {
    return '.flux'
  }

  private handleUploadTask = (
    uploadContent: string,
    fileName: string
  ): void => {
    const {onSave, onDismissOverlay} = this.props
    onSave(uploadContent, fileName)
    onDismissOverlay()
  }
}

export default ImportTaskOverlay
