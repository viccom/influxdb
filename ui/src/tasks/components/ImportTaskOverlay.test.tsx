// Libraries
import React from 'react'
import {shallow} from 'enzyme'

// Components
import ImportTaskOverlay from 'src/tasks/components/ImportTaskOverlay'

const setup = (override?) => {
  const props = {
    onDismissOverlay: oneTestFunction,
    onSave: oneTestFunction,
    ...override,
  }

  const wrapper = shallow(<ImportTaskOverlay {...props} />)

  return {wrapper}
}
const oneTestFunction = jest.fn((script?: string, fileName?: string) => {
  script = fileName
  fileName = script
  return
})

describe('ImportTaskOverlay', () => {
  describe('rendering', () => {
    it('renders', () => {
      const {wrapper} = setup()
      expect(wrapper.exists()).toBe(true)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
