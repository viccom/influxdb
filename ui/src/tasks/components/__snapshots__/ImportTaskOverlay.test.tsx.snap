// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportTaskOverlay rendering renders 1`] = `
<OverlayContainer
  maxWidth={800}
>
  <OverlayHeading
    onDismiss={[MockFunction]}
    title="Import Task"
  />
  <OverlayBody>
    <DragAndDrop
      compact={false}
      fileTypesToAccept=".flux"
      handleSubmit={[Function]}
      submitOnDrop={false}
      submitOnUpload={false}
      submitText="Upload Task"
    />
  </OverlayBody>
</OverlayContainer>
`;
